npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -155,6 +157,21 @@ export async function makeWebpackConfig (
     ],
   }
 
+  try {
```

Comment:
This is a bit of a hack. Since we cannot assume `react-dom/client` exists, we import it using a dynamic import: https://github.com/cypress-io/cypress/pull/22334/files#diff-deaf32d610fd1088b9d1a39d5b49198b98fcaba2d2b1b8942df73ed9961551e0R142

```js
import('does-not-exist').then(...).catch(...)
```

This works fine for "real" ESM, but webpack is not real ESM, but it's own implementation - it does analysis of all modules, including dynamic imports, and will error.

To tell webpack NOT to do this, we can just add it as an external - then webpack will assume it exists at runtime. We can now catch the error!

Unfortunately, this does not seem to work how you'd expect during tests - possibly due to how we use symlinks, or the fact that we ship our module is a kind of weird way (inline inside of `node_modules/cypress`.

Storybook solves this differently, using `IgnorePlugin`: https://github.com/storybookjs/storybook/pull/17215/files#diff-bc92d51dd98fe78ac7c9251c1b8d10d45bdfc8b872be68c12ace18f783205fd7R16-R18

But I could not get this to work, I am not sure why - possibly due to the weird way we bundle `cypress/react`.